p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/repository/P2RepositoryManager.java

```diff
@@ -101,6 +111,39 @@ public IArtifactRepository getCompositeArtifactRepository(Collection<Repository>
 		return new ListCompositeArtifactRepository(childs, agent);
 	}
 
+	public IArtifactRepository createLocalArtifactRepository(Path location, String name, Map<String, String> properties)
+			throws ProvisionException {
+		ArtifactRepositoryFactory factory = new SimpleArtifactRepositoryFactory();
+		factory.setAgent(agent);
+		return factory.create(location.toUri(), name, null, properties);
+	}
+
+	/**
+	 * Creates a local mirror only of the <b>data</b> of the given
+	 * {@link IMetadataRepository}, i.e. its {@code artifacts.xml/jar/xml.xz} files.
+	 */
+	public IArtifactRepository mirrorArtifactRepositoryData(IArtifactRepository repository, Path targetLocation) {
+		try {
+			IArtifactRepository mirror = createLocalArtifactRepository(targetLocation, repository.getName(),
+					repository.getProperties());
+			IStatus result = mirror.executeBatch(m -> {
+				// Copy artifacts to modifiable output repository
+				repository.query(ArtifactKeyQuery.ALL_KEYS, null).stream()
+						.map(a -> repository.getArtifactDescriptors(a)).forEach(a -> mirror.addDescriptors(a, null));
+				// Copy (potentially customized) mapping rules
+				if (repository instanceof SimpleArtifactRepository simpleSource
+						&& mirror instanceof SimpleArtifactRepository simpleTarget) {
+					simpleTarget.setRules(simpleSource.getRules()); // Copy artifact mapping rules
+				}
+			}, null);
+			StatusTool.assertNoProblemOf(IStatus.ERROR, result,
+					() -> "Failed to create output repository at: " + targetLocation);
+			return mirror;
+		} catch (ProvisionException e) {
```

Comment:
rethrow `ProvisionException`

Reply:
This makes it more complicated to use this method as it's called in a lambda BiFunction, but I can work-around that in the caller.